setup.py

```diff
@@ -4,48 +4,69 @@
-    description= ("spacebench offers benchmark datasets for causal inference, " 
-                  "incorporating real confounder and exposure data from "
-                  "environmental health studies. Synthetic outcomes are "
-                  "generated using machine learning, with spatial confounding "
-                  "introduced by masking key confounders."),
+    maintainer_email="nkhoshnevis@g.harvard.edu",
+    description=(
+        "Spatial confounding poses a significant challenge in scientific ",
+        "studies involving spatial data, where unobserved spatial ",
+        "variables can influence both treatment and outcome, possibly ",
+        "leading to spurious associations. To address this problem, ",
+        "SpaCE provides realistic benchmark datasets and tools ",
+        "for systematically evaluating causal inference methods designed to ",
+        "alleviate spatial confounding. Each dataset includes training data, ",
+        "true counterfactuals, a spatial graph with coordinates, and a ",
+        "smoothness and confounding scores characterizing the effect of a ",
+        "missing spatial confounder. The datasets cover real treatment and ",
+        "covariates from diverse domains, including climate, health and social ",
+        "sciences. Realistic semi-synthetic outcomes and counterfactuals are ",
+        "generated using state-of-the-art machine learning ensembles, following ",
+        "best practices for causal inference benchmarks. SpaCE facilitates ",
+        "an automated end-to-end machine learning pipeline, simplifying data ",
+        "loading, experimental setup, and model evaluation."
+    ),
     long_description_content_type="text/markdown",
-    entry_points={
-        "console_scripts": ["spacebench=spacebench.api.cli:main"]
-    },
-    url="https://github.com/NSAPH-Projects/space",
+    # entry_points={"console_scripts": ["spacebench=spacebench.api.cli:main"]},
+    # url="https://github.com/NSAPH-Projects/space", # cli api needs update
     license="MIT",
-    packages=find_packages(exclude=['tests*', 'scripts*', 'notebooks*'
-                                    'examples*']),
+    packages=find_packages(exclude=["tests*", "scripts*", "notebooks*" "examples*"]),
     classifiers=[
         "Development Status :: 2 - Pre-Alpha",
         "Intended Audience :: Science/Research",
         "License :: OSI Approved :: MIT License",
         "Natural Language :: English",
         "Operating System :: OS Independent",
         "Programming Language :: Python :: 3 :: Only",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: Implementation :: CPython",
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
-    python_requires='>=3.7',
+    python_requires=">=3.10",
```

Comment:
We need to drop testing 3.7, 3.8, and 3.9 in the `.github/workflows/python-app.yml` file, too.